### ISSUE TITLE
recursive_inheritance_test fails in strong mode

### ISSUE THREAD

**Issue by jmesserly** [dart-lang/sdk#26970]
we do correctly reject the code (`new MI()` not being legal in strong mode), but it seems like we're going off the rails a bit here. I would expect the error at the new expressions. Instead we seem to infer non-sensical type arguments for `new MI` and `new PMI`.

```dart
import "package:expect/expect.dart";

// Regression test for recursive inheritance patterns
abstract class Comparable<T> {
  int compare(T a);
}
class MI<T extends MI<T>> {}

class PMI<T extends Comparable<T>> extends MI<PMI<T>> {}

void main() {
  var a = new MI();
  var b = new PMI();
  // [error] A value of type 'PMI<Comparable<Comparable<T>>>' cannot be assigned to a variable of type 'MI<MI<MI<T>>>'
  a = b;
  Expect.isTrue(a is MI);
  Expect.isTrue(b is PMI);
  Expect.isTrue(b is MI);
  Expect.isTrue(b is MI<PMI>);
}
```

**Comment by jmesserly** [comment]
According to @vsmenon this is now fixed when he brings the latest Analyzer into DDC \o/

**Comment by zoechi** [comment]
@jmesserly  does this fix #25530 as well?

**Comment by jmesserly** [comment]
yeah @zoechi, I wondered the same thing! Will have to try it. In general these patterns are kind of unsound, so it may still be valid to reject the code. But I hope it rejects the code with a valid error, not because it was going off the rails and computing crazy types. :)
